controller.py

```diff
@@ -29,86 +31,46 @@ def show_applicants_list(self):
+
+            self.current_applicant = self.shortlist.applicants[i-1]
+            self.view.view_applicant_details(self.current_applicant)
             self.current_view = "applicant_details"
         except (ValueError, IndexError):
             pass
-    
+
+    def open_applicant_pdf(self):
+        startfile(self.current_applicant.cv)
 
     def run(self):
 
         self.show_boot_message()
 
         while True:
             k = readkey()
+            if k == key.ESC:
+                print("exiting the program...")
+                break
 
             if self.current_view == "home":
                 options = {"r":self.show_role_info,
                            "a":self.show_applicants_list}
                 
-                output = options.get(k)
-                if output is not None:
-                    output()
-                if k == key.ESC:
-                    print("exiting the program...")
-                    break
-                
             elif self.current_view == "applicants_list":
                 options = {"b":self.show_boot_message,
                            "d":self.show_applicant_details}
                 
-                output = options.get(k)
-                if output is not None:
-                    output()
-                if k == key.ESC:
-                    print("exiting the program...")
-                    break
-            
             elif self.current_view == "applicant_details":
                 options = {"q":self.show_applicants_list,
-                           "b":self.show_boot_message}
-
-                output = options.get(k)
-                if output is not None:
-                    output()
-
-                if k == key.ESC:
-                    print("exiting the program...")
-                    break
+                           "b":self.show_boot_message,
+                           "O":self.open_applicant_pdf}
 
             elif self.current_view == "role":
                 options = {"b":self.show_boot_message}
-
-                output = options.get(k)
-                if output is not None:
-                    output()
-
-                if k == key.ESC:
-                    print("exiting the program...")
-                    break
-
+                
             elif self.current_view == "criteria":
                 options = {"b":self.show_boot_message}
 
-                output = options.get(k)
-                if output is not None:
-                    output()
-
-                if k == key.ESC:
-                    print("exiting the program...")
-                    break
-
-"""view"""
-
-# home(default boot-up view) 
-# > "s" to shortlist view
-# > "h" for list of shortcut commands
-
-# applicant_list
-# shows only number and name
-
-# > "a" for list of applicants and their infos
-# > "c" for criteria
-# > "r" for role info and id
-# > "h" for list of shortcut commands
+            output = options.get(k)
```

Comment:
`options` needs to default to `None` in case none of the views match i.e. there are no options. Add a check `options is not None` before using it.